frontend/src/app/reducers/localizations.js

```diff
@@ -1,34 +1,60 @@
 // @flow
 
+// TODO: This Action type doesn't work below in the reducer
+// import type { Action } from 'testpilot/types';
+
 export type Localizations = {
   [language: string]: string
 };
 
-function defaultState(): Localizations {
-  return {
-  };
-}
+export type NegotiatedLanguages = Array<string>;
+
+export type LocalizationState = {
+  localizations: Localizations,
+  negotiatedLanguages: NegotiatedLanguages
+};
 
 export type SetLocalizationsAction = {
   type: 'SET_LOCALIZATIONS',
   payload: Localizations
 };
 
+export type SetNegotiatedLanguagesAction = {
+  type: 'SET_NEGOTIATED_LANGUAGES',
+  payload: NegotiatedLanguages
+};
+
+function defaultState(): LocalizationState {
+  return {
+    localizations: {},
+    negotiatedLanguages: []
+  };
+}
+
 function setLocalizations(
-  state: Localizations,
+  state: LocalizationState,
   action: SetLocalizationsAction
-): Localizations {
-  return action.payload;
+): LocalizationState {
+  return { ...state, localizations: action.payload };
+}
+
+function setNegotiatedLanguages(
+  state: LocalizationState,
+  action: SetNegotiatedLanguagesAction
+): LocalizationState {
+  return { ...state, negotiatedLanguages: action.payload };
 }
 
 export default function localizationsReducer(
-  state: ?Localizations,
-  action: SetLocalizationsAction
-): Localizations {
+  state: ?LocalizationState,
+  action: Object // TODO: figure out how to correctly assert any Action here
```

Comment:
`action: SetLocalizationsAction | SetNegotiatedLanguagesAction` should work, no? Since those are the only two types your reducer will operate on, those are the only ones you need to declare.